src/main/java/com/sandvoxel/generitech/common/tileentities/power/TestPower.java

```diff
@@ -314,4 +123,13 @@ public boolean hasCapability(Capability<?> capability, EnumFacing facing) {
 
         return super.hasCapability(capability, facing);
     }
+
+
+    @Nullable
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1199)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS106)